### PR TITLE
chore: add composite index for executions table

### PR DESCRIPTION
**Pull request by pregnantboy**
## What changed?
- This migration adds a composite index to the `executions` table that indexes `flow_id`, `test_run` and `created_at`
- This should speed up the select query for the executions page which can take quite long if the pipe has a lot of runs.

As seen here in datadog, it can take up to half a second for users with very active pipes

Migrations ran for:
- [x] UAT
- [x] Staging
- [x] Prod

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#662** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/662?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/src/db/migrations/20240702070156_add_composite_index_on_executions.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Knex } from 'knex'
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.schema.table('executions', (table) => {
+    table.index(['flow_id', 'test_run', 'created_at'])
```
Comment (ogp-weeloong) [inline packages/backend/src/db/migrations/20240702070156_add_composite_index_on_executions.ts:5]: [80% sure](https://github.com/knex/knex/issues/3913) we need `raw` if we want to build index concurrently. Our executions table is large-ish now; we should concurrently create or run this migration at 12 am :/

**Review by ogp-weeloong — APPROVED** [review]
Other than concurrent index creation LGTM. Checked that postgres query plan uses the new index for this query via `EXPLAIN`.

File: packages/backend/src/db/migrations/20240702070156_add_composite_index_on_executions.ts (hunk 1 above)
Comment (ogp-weeloong) [reply packages/backend/src/db/migrations/20240702070156_add_composite_index_on_executions.ts:5]: actually i also rmb we had to add index to `executions` table last tine (for filtering execution status on UI) and it was damn fast right. so maybe doesn't matter

**Comment by pregnantboy** [comment]
@ogp-weeloong @m0nggh need someone to shoulder check me on this

File: packages/backend/src/db/migrations/20240702070156_add_composite_index_on_executions.ts (hunk 1 above)
Comment (m0nggh) [reply packages/backend/src/db/migrations/20240702070156_add_composite_index_on_executions.ts:5]: hmm yea maybe we can do it in the evening in case

**Pull request merged**